config.go

```diff
@@ -471,6 +474,64 @@ func Dialer(d *uacp.Dialer) Option {
 // Defaults to DefaultDialTimeout. Set to zero for no timeout.
 func DialTimeout(d time.Duration) Option {
 	return func(cfg *Config) {
-		cfg.dialTimeout = &d
+		if cfg.dialer == nil {
+			cfg.dialer = &uacp.Dialer{}
+		}
+		if cfg.dialer.Dialer == nil {
+			cfg.dialer.Dialer = &net.Dialer{}
+		}
+		cfg.dialer.Dialer.Timeout = d
+	}
+}
+
+// MaxMessageSize sets the maximum message size for the UACP handshake.
+func MaxMessageSize(n uint32) Option {
+	return func(cfg *Config) {
+		if cfg.dialer == nil {
```

Comment:
maybe a small helper to avoid copy/paste of this initialization if nil code?

Reply:
Added one. Can you please check @kung-foo ?